modules/using-an-algorithm/pages/index.adoc

```diff
@@ -12,11 +12,138 @@ Each algorithm has its own folder, where you can find the implementation of the
+* CSV output to a file
+* Results written to an attribute of vertices or edges
+
+== Packaged template queries
+
+Packaged template queries are a new, simplified method for working with query algorithms that are preinstalled in a TigerGraph instance.
+They use a new keyword, `CALL`, to run with just-in-time compilation for improved performance.
+
+The syntax is very similar to using a package in Python.
+
+=== Import and explore template queries
+
+Using a TigerGraph installation version 3.8.0 and later, run the following command to import the query package.
+
+[source, gsql]
+----
+GSQL > IMPORT PACKAGE GDBMS.ALGO.*
+----
+
+Wait a few moments for the queries to load. Run `SHOW PACKAGE` to check that installation was successful.
+
+[source, gsql]
+----
+GSQL > SHOW PACKAGE
+
+Packages on global:
+  - GDBMS_ALGO
+----
+
+Run `SHOW PACKAGE` again, this time specifying `GDBMS_ALGO`, to see the list of sub-packages.
+The sub-packages correspond to a category of GDS algorithm.
+
+[source, gsql]
+----
+GSQL > SHOW PACKAGE GDBMS_ALGO
+
+Packages GDBMS_ALGO:
+  - Sub-Packages:
+    - centrality
+    - classification
+    - community
+    - graphML
+    - path
+    - similarity
+    - topological_link_prediction
+----
+
+Use dot notation to explore each sub-package.
+
+[source, gsql]
+----
+GSQL > SHOW PACKAGE GDBMS_ALGO.centrality
+
+Packages GDBMS_ALGO.centrality:
+  - Object:
+      - Packaged Queries:
+        - article_rank(string v_type, string e_type, float max_change, int maximum_iteration, float damping, int top_k, bool print_results, string result_attribute, string file_path)
+        - betweenness_cent(set<string> v_type_set, set<string> e_type_set, string reverse_e_type, int max_hops, int top_k, bool print_results, string result_attribute, string file_path, bool display_edges)
+...
+----
+
+=== Run a packaged query
+
+To run (or 'call') a packaged query, you must first be using a graph.
+Run the GSQL command `USE GRAPH <graph name>` first.
+
+Use the `CALL` command with a specific query name from a package and include the parameters.
+
+[source.wrap, gsql]
+----
+GSQL > USE GRAPH ldbc_snb
+Using graph `ldbc_snb`
+
+GSQL > CALL GDBMS_ALGO.community.louvain(["Person"],["KNOWS"],"weight",10,"","",true)
+----
+
+In this example, we call the xref:community-algorithms:louvain.adoc[] algorithm, specifying:
+
+* A set of strings `["Person"]` as the vertex types to use
+* A set of strings `["KNOWS"]` as the edge types to traverse
+* A string `"weight"` as the edge weight attribute
+* An integer `10` as the maximum limit of iterations
+* An empty string `""` for the result attribute, ensuring nothing will be written to any vertices
+* An empty string `""` for the file path, ensuring the results will not be written as a file
+* A boolean `true` to print the results to the console in JSON format.
+
+The first time you call a particular query on a graph, it goes through a short installation process.
+This only needs to happen once per query per graph.
```

Comment:
If you run the same algorithm with the same schema elements (but other parameters can be modified), it will skip the compilation and immediately run the algorithm query.

Reply:
It might be helpful to have some examples.
1. Running an algorithm for the first time, with a certain set of parameters.  Mention that compilation will occur.
2. Running the same algorithm with the same schema elements but changes in other parameters. Mention that compilation will not occur.